model/variable.cc

```diff
@@ -415,6 +415,10 @@ bool VariableBase::visible() const
   auto g=group.lock();
   //toplevel i/o items always visible
   if ((!g || !g->group.lock()) && g==controller.lock()) return true;
+  // ensure vars attached to invisible ops are also made invisible. for ticket 1275
+  if (!ports[0]->wires().empty())
+    if (auto i=ports[0]->wires()[0]->to()->item().operationCast())
+      if (!i->visible()) return false;
```

Comment:
This doesn't sound right. If a variable has been moved to the var tab, then it should still be visible if downstream items are visible.

An item becomes invisible if all downstream items are invisible or a downstream item is  variable move to the variable (regardless of visibility).

These rules need to be applied recursively, of course.

Reply:
> This doesn't sound right. If a variable has been moved to the var tab, then it should still be visible if downstream items are visible.

Doesn't the code I added only make the var invisible if an op attached to its outport is invisible by virtue of another downstream var already being invisible?

I think my comment above the code may be misleading. I meant to say "ensure vars with invisible ops attached to their outports are also made invisible. for ticket 1275"

Reply:
Doesn't the code I added only make the var invisible if an op attached to its outport is invisible by virtue of another downstream var already being invisible?

Yes - but this is not what you want to do. A variable is invisible if it moved to the variable tab and no item attached to its output port is visible

Reply:
So, the variable itself should stay visible but not the items attached to it? I quote from ticket 72:

> The defining groupm should then be removed from the canvas (or made invisible).